crates/provider-http-server/src/lib.rs

```diff
@@ -365,6 +365,18 @@ impl HttpServerCore {
 
         let task_handle = handle.clone();
         let target = target.to_owned();
+        let socket = tokio::net::TcpSocket::new_v4().context("Unable to open socket")?;
```

Comment:
What about IPv6 addresses? It seems we should have a conditional here using the right socket depending on `addr`.
btw, we should probably listen on IPv6 by default if IP is not specified, since that normally binds IPv4 as well

Reply:
Ah good point! Let me go fix that

Reply:
I left our current IPv4 default for now to keep it small, but we can go back and fix it later as well